src/algebra/order/floor.lean

```diff
@@ -96,8 +96,13 @@ lemma le_floor (h : (n : α) ≤ a) : n ≤ ⌊a⌋₊ := (le_floor_iff $ n.cast
 
 lemma floor_lt (ha : 0 ≤ a) : ⌊a⌋₊ < n ↔ a < n := lt_iff_lt_of_le_iff_le $ le_floor_iff ha
 
+lemma floor_lt_one (ha : 0 ≤ a) : ⌊a⌋₊ < 1 ↔ a < 1 :=
+(floor_lt ha).trans $ by rw nat.cast_one
+
 lemma lt_of_floor_lt (h : ⌊a⌋₊ < n) : a < n := lt_of_not_le $ λ h', (le_floor h').not_lt h
 
+lemma lt_one_of_floor_lt_one (h : ⌊a⌋₊ < 1) : a < 1 := by exact_mod_cast lt_of_floor_lt h
```

Comment:
About all additions in this file, do you plan on writing a lemma version for each natural? 😛 

I believe the new cast will make that better because `1 : α` will really be `(1 : ℕ) : α`.

Reply:
I agree, the need for these largely disappears once we refactor nat casts

Reply:
> do you plan on writing a lemma version for each natural? 

I was initially thinking that I could do it for `bit0` and `bit1`, but I'm not sure if those are all actually true.

Reply:
I'm not quite happy with this duplication but I guess I will wait until `2` to start shouting.